src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -308,52 +329,82 @@ private List<Piece> getSuccessfullyProcessedPieces(String poLineId, Map<String,
     return pieceStorageService.getPieces(Integer.MAX_VALUE, 0, query, requestContext)
       .map(PieceCollection::getPieces);
   }
 
+  private PoLine updateRelatedPoLineDetails(PoLine poLine,
+                                            List<Piece> piecesFromStorage,
+                                            List<Piece> byPoLine,
+                                            List<Piece> successfullyProcessed) {
+    ReceiptStatus receiptStatus = calculatePoLineReceiptStatus(byPoLine, successfullyProcessed, poLine);
+    purchaseOrderLineService.updatePoLineReceiptStatusWithoutSave(poLine, receiptStatus);
+
+    // the same check as in PieceUpdateFlowManager::updatePoLine
+    if (Boolean.TRUE.equals(poLine.getIsPackage()) || Boolean.TRUE.equals(poLine.getCheckinItems())) {
+      logger.info("updateRelatedPoLineDetails:: Skipping updating POL '{}' if it package or has independent receiving flow", poLine.getId());
+      return poLine;
+    }
+
+    for (Piece pieceToUpdate : successfullyProcessed) {
+      Optional<Piece> relatedStoragePiece = piecesFromStorage.stream()
+        .filter(piece -> piece.getId().equals(pieceToUpdate.getId()))
+        .findFirst();
+      if (relatedStoragePiece.isEmpty()) {
+        continue;
+      }
+      CompositePoLine compositePoLine = PoLineCommonUtil.convertToCompositePoLine(poLine);
+      PieceUpdateHolder holder = new PieceUpdateHolder();
+      holder.withPieceFromStorage(relatedStoragePiece.get());
+      holder.withPieceToUpdate(pieceToUpdate);
+      holder.withPoLineOnly(compositePoLine);
+      pieceUpdateFlowPoLineService.updatePoLineWithoutSave(holder);
+      return HelperUtils.convertToPoLine(compositePoLine);
+    }
+    return poLine;
+  }
+
   /**
    * Returns "Fully Received" status if quantity of expected piece records is
    * zero, otherwise checks how many received pieces. If quantity of received
    * piece records is zero, returns "Awaiting Receipt" status, otherwise -
    * "Partially Received"
    *
-   * @param byPoLine all piece records obtained by PO line id
-   * @param poLine   PO Line record representation from storage
+   * @param piecesByPoLine               all piece records obtained by PO line id
+   * @param piecesSuccessfullyProcessed  pieces that were successfully processed
+   * @param poLine                       PO Line record representation from storage
    * @return completable future holding calculated PO Line's receipt status
    */
-  private Future<ReceiptStatus> calculatePoLineReceiptStatus(List<Piece> byPoLine, PoLine poLine, List<Piece> pieces) {
-    long expectedPiecesQuantity = byPoLine.stream()
+  private ReceiptStatus calculatePoLineReceiptStatus(List<Piece> piecesByPoLine, List<Piece> piecesSuccessfullyProcessed, PoLine poLine) {
+    logger.info("calculatePoLineReceiptStatus:: piecesByPoLine: {}", piecesByPoLine.size());
+    logger.info("calculatePoLineReceiptStatus:: pieces: {}", piecesSuccessfullyProcessed.size());
```

Comment:
lets add meaningful messages, so it can be usefuly to understand flow

Reply:
as suggestion, 
```
logger.info("calculatePoLineReceiptStatus:: Trying to calculate poLineReceiptStatus with id '{}' using '{}' piecesByPoLine and '{}' piece(s) and", poLine.getId(), piecesByPoLine.size(), piecesSuccessfullyProcessed.size())
```

Reply:
These were my diagnostic logs, will either remove or modify them